addons/stock/migrations/8.0.1.1/pre-migration.py

```diff
@@ -0,0 +1,164 @@
+#              (C) 2014 Therp B.V.
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+import logging
+from openerp.openupgrade import openupgrade
+
+
+logger = logging.getLogger('OpenUpgrade.stock')
+
+column_renames = {
+    'product_product': [
+        ('track_incoming', None),
+        ('track_outgoing', None),
+        ('track_production', None),
+        ('valuation', None)
+    ],
+    'stock_inventory': [
+        ('date_done', None),
+    ],
+    'stock_inventory_line': [
+        ('product_uom', 'product_uom_id'),
+    ],
+    'stock_move': [
+        ('auto_validate', None),
+        ('price_currency_id', None),
+        ('prodlot_id', None),
+        ('product_qty', 'product_uom_qty'),
+        ('tracking_id', None)
+    ],
+    'stock_production_lot': [
+        ('company_id', None),
+        ('create_date', None),
+        ('date', 'create_date'),
+        ('prefix', None)
+    ],
+    'stock_picking': [
+        ('type', None),
+    ],
+    'stock_location': [
+        ('icon', None),
+        ('chained_journal_id', None),
+        ('chained_location_id', None),
+        ('chained_location_type', None),
+        ('chained_auto_packing', None),
+        ('chained_company_id', None),
+        ('chained_delay', None),
+        ('chained_picking_type', None),
+    ],
+    'stock_warehouse': [
+        ('lot_input_id', 'wh_input_stock_loc_id'),
+        ('lot_output_id', 'wh_output_stock_loc_id'),
+    ],
+    'stock_warehouse_orderpoint': [
+        ('product_uom', None),
+    ],
+}
+
+xmlid_renames = [
+    ('procurement.sequence_mrp_op_type', 'stock.sequence_mrp_op_type'),
+    ('procurement.sequence_mrp_op', 'stock.sequence_mrp_op'),
+    ('stock.property_stock_account_input_categ',
+        'stock_account.property_stock_account_input_categ'),
+    ('stock.property_stock_account_input_prd',
+        'stock_account.property_stock_account_input_prd'),
+    ('stock.property_stock_account_output_categ',
+        'stock_account.property_stock_account_output_categ'),
+    ('stock.property_stock_account_output_prd',
+        'stock_account.property_stock_account_output_prd'),
+    ('stock.property_stock_journal', 'stock_account.property_stock_journal'),
+    ('stock.stock_journal', 'stock_account.stock_journal'),
+]
```

Comment:
I'm not sure.
Upgrade of 'stock' will delete xml_ids that disappeared. (except noupdate xml_id)
When process will update 'stock_account', the object has been deleted.

Regards.